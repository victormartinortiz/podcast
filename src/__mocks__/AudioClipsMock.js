const AudioClipsMock = [
  {
    id: 6751040,
    title: 'Posta Originals: Guerra sin fin',
    description:
      'Se cumplen siete años de la guerra en Siria, la peor crisis humanitaria de nuestros tiempos, con más de 400 mil muertos, 6 millones de refugiados y unos 9 millones de desplazados internos. ¿Cómo fue que los levantamientos de la Primavera Árabe en 2011 terminaron en esta guerra civil? ¿Cómo es vivir en un país en guerra?\r\n\r\nEste primer documental de Posta Originals, conducido y producido por la periodista Ayelén Oliva, va detrás de los intereses que motivaron y continúan alimentando el conflicto y se pregunta por su posible solución. Música compuesta por Kevin MacLeod, usada bajo licencia Creative Commons: By Attribution 3.0\\. Imagen de portada: Tammam Azzam. Edición: Luciano Banchero.\r\n\r\nUn podcast original de Posta.',
    updated_at: '2018-05-04T15:01:47.000Z',
    user: {
      id: 4689293,
      username: 'lucianobanchero',
      urls: {
        profile: 'https://audioboom.com/lucianobanchero',
        image: 'https://images.theabcdn.com/i/21351474',
        profile_image: {
          original: 'https://images.theabcdn.com/i/21351474'
        }
      }
    },
    link_style: 'channel',
    channel: {
      id: 4702115,
      title: 'POSTA',
      urls: {
        detail: 'https://audioboom.com/channel/postafm',
        logo_image: {
          original: 'https://images.theabcdn.com/i/28456821'
        }
      }
    },
    duration: 1762.86,
    mp3_filesize: 28773734,
    uploaded_at: '2018-03-26T18:31:44.000Z',
    recorded_at: '2018-03-26T18:31:44.000+00:00',
    uploaded_at_ts: 1522089104,
    recorded_at_ts: 1522089104,
    can_comment: false,
    can_embed: true,
    category_id: 282,
    counts: {
      comments: 0,
      likes: 4,
      plays: 2695
    },
    urls: {
      detail:
        'https://audioboom.com/posts/6751040-posta-originals-guerra-sin-fin',
      high_mp3:
        'https://audioboom.com/posts/6751040-posta-originals-guerra-sin-fin.mp3',
      image: 'https://images.theabcdn.com/i/32177134',
      post_image: {
        original: 'https://images.theabcdn.com/i/32177134'
      },
      wave_img: 'https://images.theabcdn.com/i/w/7791829'
    },
    image_attachment: 32177134
  }
];

export default AudioClipsMock;
