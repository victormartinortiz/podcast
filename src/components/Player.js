import React from 'react';
import Link from 'next/link';
import '../assets/styles/components/Player.scss';

const Player = ({ clip }) => (
  <>
    <div className="modal">
      <div className="clip">
        <nav>
          <Link href={`/channel?id=${clip.channel.id}`}>
            <a className="close">&lt; Volver</a>
          </Link>
        </nav>

        <picture>
          <div
            style={{
              backgroundImage: `url(${clip.urls.image ||
                clip.channel.urls.logo_image.original})`
            }}
          />
        </picture>

        <div className="player">
          <h3>{clip.title}</h3>
          <h6>{clip.channel.title}</h6>
          <audio controls>
            <source src={clip.urls.high_mp3} type="audio/mpeg" />
          </audio>
        </div>
      </div>
    </div>
  </>
);

export default Player;