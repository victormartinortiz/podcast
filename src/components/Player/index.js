/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { Link } from '../../../routes';
import slug from '../../utils/slug';
import {
  Modal,
  Episode,
  Nav,
  NavLink,
  Picture,
  Image,
  PlayerControl,
  PlayerTitle,
  PlayerSubTitle,
  PlayerAudio
} from './styles';

const Player = ({ clip, onClose }) => (
  <>
    <Modal>
      <Episode>
        <Nav>
          {onClose ? (
            <NavLink onClick={onClose}>&lt; Volver </NavLink>
          ) : (
            <Link
              route='channel'
              params={{ slug: slug(clip.channel.title), id: clip.channel.id }}
            >
              <NavLink className='close'>&lt; Volver</NavLink>
            </Link>
          )}
        </Nav>
        <Picture>
          <Image src={clip.urls.image} alt='episodeimage' />
        </Picture>
        <PlayerControl>
          <PlayerTitle>{clip.title}</PlayerTitle>
          <PlayerSubTitle>{clip.channel.title}</PlayerSubTitle>
          <PlayerAudio controls>
            <source src={clip.urls.high_mp3} type='audio/mpeg' />
          </PlayerAudio>
        </PlayerControl>
      </Episode>
    </Modal>
  </>
);

export default Player;
