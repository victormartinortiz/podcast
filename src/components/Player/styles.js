/* eslint-disable import/prefer-default-export */
import styled from 'styled-components';

export const Modal = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 99999;
`;

export const Episode = styled.div`
  display: flex;
  height: 100%;
  flex-direction: column;
  background: #8756ca;
  color: white;
`;

export const Nav = styled.nav`
  background: none;
`;

export const NavLink = styled.a`
  display: inline-block;
  padding: 15px;
  color: white;
  cursor: pointer;
  text-decoration: none;
`;

export const Picture = styled.picture`
  display: flex;
  align-items: center;
  justify-content: center;
  flex: 1 1;
  flex-direction: column;
  width: auto;
  padding: 10%;
`;

export const Image = styled.image`
  width: 100%;
  height: 100%;
  background-position: 50% 50%;
  background-size: contain;
  background-repeat: no-repeat;
  background-image: url(${props => props.src})
`;

export const PlayerControl = styled.div`
  padding: 30px;
  background: rgba(0, 0, 0, 0.3);
  text-align: center;
`;

export const PlayerTitle = styled.h3`
  margin: 0;
`;

export const PlayerSubTitle = styled.h6`
  margin: 0;
  margin-top: 1em;
`;

export const PlayerAudio = styled.audio`
  margin-top: 2em;
  width: 100%;
`;
