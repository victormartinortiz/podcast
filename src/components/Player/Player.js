import React from 'react';
import { Link } from '../../routes';
import slug from '../utils/slug';
import '../assets/styles/components/Player.scss';

const Player = ({ clip, onClose }) => (
  <>
    <div className="modal">
      <div className="clip">
        <nav>
        { onClose ?
          <a onClick={onClose}>&lt; Volver</a>
          :
          <Link
            route="channel"
            params={{ slug: slug(clip.channel.title), id: clip.channel.id }}
          >
            <a className="close">&lt; Volver</a>
          </Link>
        }
        </nav>

        <picture>
          <div
            style={{
              backgroundImage: `url(${clip.urls.image ||
                clip.channel.urls.logo_image.original})`
            }}
          />
        </picture>

        <div className="player">
          <h3>{clip.title}</h3>
          <h6>{clip.channel.title}</h6>
          <audio controls>
            <source src={clip.urls.high_mp3} type="audio/mpeg" />
          </audio>
        </div>
      </div>
    </div>
  </>
);

export default Player;
