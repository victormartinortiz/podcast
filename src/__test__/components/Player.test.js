import React from 'react';
import { shallow, mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Player from '../../components/Player';
import AudioClipsMock from '../../__mocks__/AudioClipsMock';

describe('<Player />', () => {
  it('renders Player unchanged', () => {
    const tree = create(<Player clip={AudioClipsMock[0]} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render Player component', () => {
    const player = shallow(<Player clip={AudioClipsMock[0]} />);
    expect(player.length).toEqual(1);
  });

  it('check onClose modal', () => {
    const onClose = jest.fn()
    const wrapper = mount(<Player clip={AudioClipsMock[0]} onClose={onClose}/>)
    wrapper.find('a').simulate('click')
    expect(wrapper.find('.close')).toHaveLength(0)
  })
});
