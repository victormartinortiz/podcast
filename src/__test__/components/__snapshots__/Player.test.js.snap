// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> renders Player unchanged 1`] = `
<div
  className="styles__Modal-o7thus-0 cjCqka"
>
  <div
    className="styles__Episode-o7thus-1 bNtjbq"
  >
    <nav
      className="styles__Nav-o7thus-2 jQFgKT"
    >
      <a
        className="styles__NavLink-o7thus-3 fGiZyr close"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        &lt; Volver
      </a>
    </nav>
    <picture
      className="styles__Picture-o7thus-4 dSZxxn"
    >
      <div
        alt="episode image"
        className="styles__Image-o7thus-5 gNtvUn"
        src="https://images.theabcdn.com/i/32177134"
      />
    </picture>
    <div
      className="styles__PlayerControl-o7thus-6 cpBbxy"
    >
      <h3
        className="styles__PlayerTitle-o7thus-7 cbiuLu"
      >
        Posta Originals: Guerra sin fin
      </h3>
      <h6
        className="styles__PlayerSubTitle-o7thus-8 cFDamJ"
      >
        POSTA
      </h6>
      <audio
        className="styles__PlayerAudio-o7thus-9 BgpTR"
        controls={true}
      >
        <source
          src="https://audioboom.com/posts/6751040-posta-originals-guerra-sin-fin.mp3"
          type="audio/mpeg"
        />
      </audio>
    </div>
  </div>
</div>
`;
